reconcile/utils/terrascript_aws_client.py

```diff
@@ -4788,6 +4788,9 @@ def populate_tf_resource_alb(self, spec, ocm_map=None):
             target_name = t["name"]
             t_openshift_service = t.get("openshift_service")
             t_ips = t.get("ips")
+            t_protocol = t.get("protocol", "HTTPS")
+            t_protocol_version = t.get("protocol_version", "HTTP1")
+
```

Comment:
```python
a = {"a": None}
print(a.get("a", "default"))
# --> None

print(a.get("a") or "default")
# --> default
```

According to schema we have an optional value. `get()` can then actually return that `None` value. To work around that, we usually do the `or` syntax instead when we have an Optional as the value in a dict. But I agree that this also has its pitfalls as you outlined.

My understanding is that in the case of `.get(x, default)` you will require an additional check to make sure there is no `None` returned.

```python
t_protocol = "HTTPS" if t_protocol is None
```

Usually mypy would already complain here - but the terrascript file is not checked by mypy if im not mistaken - so this goes unnoticed.

Reply:
@fishi0x01 

I'm not sure the internals of `qontract-reconcile` but what you just shared means that if I don't set `protocol` explicitly in the resource YAML, it will default to `None` instead of the field not being preset i.e 

```
t['protocol'] is None
```

as opposed to
 
```
'protocol' in t is false
```